tempodb/search/pipeline.go

```diff
@@ -91,6 +87,54 @@ func NewSearchPipeline(req *tempopb.SearchRequest) Pipeline {
 	return p
 }
 
+// rewriteTagLookup intercepts certain tag/value lookups and rewrites them. It returns
+// true if the tag lookup should be excluded from the remaining tag/value lookups because
+// the it was rewritten into a different filter altogehter.  Otherwise it returns false,
+// and a new set of tag/value strings to use, which will either be the original inputs
+// or rewritten lookups.
+func (p *Pipeline) rewriteTagLookup(k, v string) (skip bool, newk, newv string) {
+	switch k {
+	case SecretExhaustiveSearchTag:
+		// Perform an exhaustive search by:
+		// * no block or page filters means all blocks and pages match
+		// * substitute this trace filter instead rejects everything. therefore it never
+		//   quits early due to enough results
+		p.tracefilters = append(p.tracefilters, func(s tempofb.Trace) bool {
+			return false
+		})
+		// Skip
+		return true, "", ""
+
+	case ErrorTag:
+		// Convert error=true|false into status.code=2|1
+		b, err := strconv.ParseBool(v)
+		if err != nil {
+			// Not a bool string = fall-through
+			break
+		}
+
+		if b {
+			// Error = true
+			return false, StatusCodeTag, strconv.Itoa(int(v1.Status_STATUS_CODE_ERROR))
+		} else {
+			// Error = false
+			return false, StatusCodeTag, strconv.Itoa(int(v1.Status_STATUS_CODE_OK))
```

Comment:
I'm worried this will be confusing: OTel only sets STATUS_CODE_OK if someone explicitly marks the operation as successful. By default every operation that is not an error gets STATUS_CODE_UNSET.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status

I'd expect `error=false` to mean STATUS_CODE_UNSET _or_ STATUS_CODE_OK, but we can't express that yet 😞  But this means that searching `error=false` will most likely not do what you expect it to do.

Not sure what the best alternative is though...

Reply:
Ah ok thanks for clarifying this behavior. I noticed it in our test data but didn't realize the convention.  Agree we can't express it yet 😞 Let's keep thinking and discussing.